src/librustdoc/html/render.rs

```diff
@@ -246,6 +247,56 @@ struct IndexItem {
     path: String,
     desc: String,
     parent: Option<ast::DefId>,
+    search_type: Option<IndexItemFunctionType>,
+}
+
+/// A type used for the search index.
+struct Type {
+    name: Option<String>,
+}
+
+impl fmt::Display for Type {
+    /// Formats type as {name: $name}.
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        try!(write!(f, "{{"));
+        // Wrapping struct fmt should never call us when self.name is None,
+        // but just to be safe we write `null` in that case.
+        try!(write!(f, "\"name\":\"{}\"", self.name.as_ref().unwrap_or(&"null".to_string())));
```

Comment:
If the name is `None`, I think this will output `"null"` instead of `null` (which may confuse the JS)